sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -290,11 +290,6 @@ object TypeCoercion {
       // Skip nodes who's children have not been resolved yet.
       case e if !e.childrenResolved => e
 
-      case a @ BinaryArithmetic(left @ StringType(), right @ DecimalType.Expression(_, _)) =>
-        a.makeCopy(Array(Cast(left, DecimalType.SYSTEM_DEFAULT), right))
-      case a @ BinaryArithmetic(left @ DecimalType.Expression(_, _), right @ StringType()) =>
-        a.makeCopy(Array(left, Cast(right, DecimalType.SYSTEM_DEFAULT)))
-
```

Comment:
Hi, @dilipbiswal .
IMHO, the root cause seems to be **decimal multiplication** between `decimal(38,18)`s.
```
scala> sql("select cast(10 as decimal(38,18)) * cast(10 as decimal(38,18))").head
res0: org.apache.spark.sql.Row = [null]
```
How do you think about this?